paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnwareBucketCompactionHelper.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink.compact;
+
+import org.apache.paimon.annotation.VisibleForTesting;
+import org.apache.paimon.append.AppendOnlyCompactionTask;
+import org.apache.paimon.flink.sink.Committable;
+import org.apache.paimon.operation.AppendOnlyFileStoreWrite;
+import org.apache.paimon.table.FileStoreTable;
+import org.apache.paimon.table.sink.CommitMessage;
+import org.apache.paimon.table.sink.TableCommitImpl;
+
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * The help class for the Compaction of unware bucket append table to execute {@link
+ * AppendOnlyCompactionTask}.
+ */
+public class UnwareBucketCompactionHelper {
+    private final FileStoreTable table;
+    private final String commitUser;
+
+    private final transient AppendOnlyFileStoreWrite write;
+
+    protected final transient Queue<Future<CommitMessage>> result;
+
+    private final transient Supplier<ExecutorService> compactExecutorsupplier;
+
+    public UnwareBucketCompactionHelper(
+            FileStoreTable table,
+            String commitUser,
+            Supplier<ExecutorService> lazyCompactExecutor) {
+        this.table = table;
+        this.commitUser = commitUser;
+        this.write = (AppendOnlyFileStoreWrite) table.store().newWrite(commitUser);
+        this.result = new LinkedList<>();
+        this.compactExecutorsupplier = lazyCompactExecutor;
+    }
+
+    public void processElement(StreamRecord<AppendOnlyCompactionTask> element) throws Exception {
```

Comment:
`public void submit(AppendOnlyCompactionTask task) throws Exception`?  Compactor does not need to access `StreamRecord`, just pass `AppendOnlyCompactionTask` to it is okay.

Reply:
done